ruby_programming/git/lesson_a_deeper_look_at_git.md

```diff
@@ -42,3 +42,20 @@ This section contains helpful links to other content. It isn't required, so cons
 * Watch [Get Going from Pro Git](http://git-scm.com/video/get-going) if you'd like a refresher of things.
 * Read [Git Cheat Sheet](https://www.atlassian.com/git/tutorials/atlassian-git-cheatsheet) if you need a reference sheet.
 * Watch [Using Rebase & Merge](https://www.youtube.com/watch?v=f1wnYdLEpgI) for an example on how to use both rebase and merge.
+
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
+
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Basics-Undoing-Things'>How do you amend a commit message that you just messed up?</a>
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Basics-Viewing-the-Commit-History'>How do you view the history of your most recent commits?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell#_switching_branches'>What are two different uses for `$ git checkout`?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Basics-Undoing-Things#_unstaging'>How do you undo a recent commit?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell'>What are branches?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell#_create_new_branch'>How do you create a new branch to work on?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Remote-Branches#_pushing_branches'>How do you push that (non-main) branch up to GitHub?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging#_basic_merging'>How do you merge the changes from your new branch into the main branch again?</a> 
+* <a class='knowledge-check-link' href='https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell'>Why is working with feature branches useful?</a>
```

Comment:
I'm not seeing where `feature branches` are explicitly mentioned as a concept here, can you point that out to me as well?

Reply:
The term _feature branch_ is not mentioned anywhere in the git book provided as a resource. I am not really sure. I will post another comment with a possible solution because it is related to this.

Reply:
@wise-king-sullyman  Hey there, I have been thinking about limiting the number of knowledge checks and I have provided the ones I think might be removed and also my thought process on why I think it should be done. Please let me know of you opinion as well :D. If you agree, the learning outcomes should be changed as well to match these checks, right?
* What is the difference between a "remote" and your local repo?
	 Reason: People can understand the difference between the two from the git basics foundations lesson when they clone a repository


* How do you view the history of your most recent commits?
	Reason : Git log is covered as well in the Foundations lesson



* What are branches and why is working with branches useful?
	I would suggest merging these two questions. Reading the whole linked page should cover both. Both questions had exactly the same link.

* How do you add your GitHub repo as the remote?
	Reason: This might not be that crucial to post, it can be easily googled as the student knows by now the difference between local and remote. Finding how to do it should be a piece of cake
	
* How do you push that (non-main) branch up to GitHub?
	Reason: Once the student understands the notion of branches, pushing a non main branch is a trivial thing to do. For this reason, this can be removed as well I think.

Reply:
> If you agree, the learning outcomes should be changed as well to match these checks, right?

Ah, I missed that you used the learning outcomes as the knowledge checks. That's actually not something we want to do, but I think most of the current Learning Outcomes are better actually better as Knowledge Checks anyways so it will probably make more sense to also rework the Learning Outcomes while you're working on this lesson if you'd like to do that.

Check the wiki post I linked at the bottom of the issue for the explanation of why we don't want them to be the same, and what we want to see different between them.

I do agree that your suggested questions to remove can probably all go :+1: